sopel/bot.py

```diff
@@ -587,32 +587,48 @@ def call_rule(
         sopel: 'SopelWrapper',
         trigger: Trigger,
     ) -> None:
+        nick = trigger.nick
+        context = trigger.sender
+        is_channel = context and not trigger.is_privmsg
+
         # rate limiting
         if not trigger.admin and not rule.is_unblockable():
-            if rule.is_rate_limited(trigger.nick):
+            if rule.is_rate_limited(nick):
+                message = rule.get_user_rate_message(nick)
+                if message:
+                    sopel.notice(message, destination=nick)
                 return
-            if not trigger.is_privmsg and rule.is_channel_rate_limited(trigger.sender):
+
+            if is_channel and rule.is_channel_rate_limited(context):
+                message = rule.get_channel_rate_message(nick, context)
+                if message:
+                    sopel.notice(message, destination=nick)
                 return
+
             if rule.is_global_rate_limited():
+                message = rule.get_global_rate_message()
+                if message:
+                    sopel.notice(message, destination=nick)
                 return
 
         # channel config
-        if trigger.sender in self.config:
-            channel_config = self.config[trigger.sender]
+        if context in self.config:
```

Comment:
Should this be `if is_channel and context in self.config`? I realize it's already possible to make a "Channel configuration" for some specific nick because of how this works, but the question is whether that's a bug/needs changing. Triggering this section of code in PM from someone whose nick matches a plugin's section name, for instance, feels squicky.

Reply:
Yeah I know, it's weird. I don't know how to feel about that. I remember that the reminder plugin didn't mind using "sender" even in PM. So I don't know yet.

Reply:
Fixed.